[Identity] Update README docs to reflect Azure.Core 1.53 type consolidation (#58082)
Azure/azure-sdk-for-net: sdk/core/Azure.Core/README.md

- Exceptions for reporting errors from service requests in a consistent fashion. (`RequestFailedException`).
- Customizing requests (`RequestContext`).
- Abstractions for representing Azure SDK credentials. (`TokenCredentials`).

> - Authenticating Azure SDK clients using credential types from Microsoft Entra ID without a separate package dependency. (`DefaultAzureCredential`, `ManagedIdentityCredential`, and all other credential types from `Azure.Identity` are included directly in `Azure.Core` starting with version 1.53.)

Reviewer: ```suggestion
- Authenticating Azure SDK clients to Microsoft Entra ID using token-based credential types without a separate package dependency. `DefaultAzureCredential`, `ManagedIdentityCredential`, and all other credential types from the `Azure.Identity` package are included in `Azure.Core`, starting with version 1.53.0.
```

More on mocking in [Unit testing and mocking with the Azure SDK for .NET](https://learn.microsoft.com/dotnet/azure/sdk/unit-testing-mocking).

### Authenticating Service Clients

> Starting with Azure.Core 1.53, all credential types previously in `Azure.Identity` are available directly:

Reviewer 2: ```suggestion
Starting with Azure.Core 1.53.0, all credential types previously in `Azure.Identity` are available directly from the Azure service client libraries, which all take a dependency on `Azure.Core`:
```

```C# Snippet:AzureCoreDefaultCredential

> // No Azure.Identity package reference required for Azure.Core 1.53+

Reviewer 2: ```suggestion
// No Azure.Identity package reference required for client SDKs that depend on Azure.Core 1.53.0+
```

Reviewer 2: Why did we change this back? The intent of this change is to make it clear that developers are more likely to get their Azure.Core reference from the client SDK they are using.

Author: This was a mistake, probably got overridden when I ran the update snippets script.
Fixed now :)
